scripts/installer/utils/extractors.py

```diff
@@ -1,161 +1,52 @@
-                extract_ds_object = True
-                key = [x.strip() for x in line.split("=", 1)][0]
-                data_studio_options[key] = {}
-                indices_to_pop.append(i)
-                current_key = key
-            # Each value of a DS object
-            elif (extract_ds_object) and ("=" in line):
-                key, value = [x.strip() for x in line.split("=", 1)]
-                data_studio_options[current_key][key] = value.strip('"')
-                indices_to_pop.append(i)
-            elif (extract_ds_object) and ("}," in line):
-                extract_ds_object = False
-                current_key = None
-            elif (start_handling_ds_options) and (line == "}"):
-                indices_to_pop.append(i)
-                break
-
-        purge_indices_in_reverse(indices_to_pop)
-        data["data_studio_options"] = data_studio_options
-
-        # 4) Handle multiline arrays. Find opening line with '=' and ending in '['
-        target_index = None
-        indices_to_pop = []
-        for i, line in enumerate(lines_array):
-            if ("=" in line) and (line.strip()[-1] == "["):
-                target_index = i
-                continue
-
-            if target_index is not None:
-                lines_array[target_index] += line.strip()
-                indices_to_pop.append(i)
-
-            if line.strip()[-1] == "]":
-                target_index = None
-
-        purge_indices_in_reverse(indices_to_pop)
-
-        # 5) Convert items to proper python types.
-        for line in lines_array:
-            if "=" in line:
-                key, value = [x.strip() for x in line.split("=", 1)]
-                if value.lower() == "true":
-                    data[key] = True
-                elif value.lower() == "false":
-                    data[key] = False
-                else:
-                    data[key] = ast.literal_eval(value)
-
-    return data
-
-    # # with open('output.json', 'w') as file:
-    # #     json.dump(data, file, indent=4)
-
-
 def get_tfvars_as_json():
-    return convert_tfvars_to_dictionary("terraform.tfvars")
+    """Uses the `tmccombs/hcl2json` Docker image to convert `terraform.tfvars` into JSON format and return it as a Python dictionary."""
+    
+    tfvars_path = os.path.abspath("terraform.tfvars")
+    timestamp = datetime.now().strftime("%Y_%b%d_%I-%M%p")
+    
+    if not os.path.exists(tfvars_path):
+        raise FileNotFoundError("terraform.tfvars file not found in current directory.")
+
+    try:
+        # Create a temporary file for Docker output
+        # delete=True would delete the tmp file as soon as the `with` block exits but we need the file further down
+        with tempfile.NamedTemporaryFile(delete=False, prefix=f"terraform_tfvars_{timestamp}_", suffix=".json", dir="/tmp") as temp_output:
+            output_path = temp_output.name
+
+        cmd = [
+                "docker", "run",
+                "--platform", "linux/amd64",
```

Comment:
Are we making an assumption about the execution OS? 

It would be `amd64` for me due to WSL usage but I think it would be `arm64` for a subset of Macs. (_I forget the exact language I had re: expected OS platform -- I don't particularly relish having to juggle a configuration for the machine type executing the TF solution but this just might be a necessary cost to avoid their need to install Golang_).

Reply:
Handled by adding a bit of code to dynamically check chip-type at runtime.